### ISSUE TITLE
Establish editorial rules for pattern title and description

### ISSUE THREAD

**Issue by afercia** [WordPress/pattern-directory#572]
**Describe the bug**
The patterns title and description do have an impact on accessibility and usability of the WP user interface. They do have an impact also on the Patterns directory SEO. Their text should be meaningful and should avoid repetition. Also, the title and description used for the user interface are arguably ideal for SEO purposes.

I'm not sure this can be solved on the code side. Instead, I'd tend to think some editorial rules and best practices should be established in order to craft meaningful, useful, titles and descriptions. Also, it is worth considering rules for the patterns review process.

A few examples:

## From an accessibility perspective:

[This pattern](https://wordpress.org/patterns/pattern/three-columns-of-text/) title and description use the same text:
```
Title:       Three columns of text
Description: Three columns of text.
```

The only difference is a period at the end of the description.  
When navigating through the list of patterns in the Inserter, screen readers announce both the title and the (visually hidden) description:

<img width="951" alt="Screenshot 2023-03-29 at 11 46 16" src="https://user-images.githubusercontent.com/1682452/228532885-e39aa477-e6fa-424c-9655-3cbc058d55d4.png">

That's redundant and just adds unnecessary noise. 

For [this other pattern](https://wordpress.org/patterns/pattern/three-columns-with-images-and-text/), the first part of the description repeats the title text and then adds some more information:
```
Title:       Three columns with images and text
Description: Three columns with images and text, with vertical spacing for an offset look.
```

Also in this case, the repetition should be avoided, as the complete text announced by screen reader is noisy and redundant:

<img width="1084" alt="Screenshot 2023-03-29 at 11 18 01" src="https://user-images.githubusercontent.com/1682452/228535307-13f8bcc1-ee0a-4310-a337-f4831effaa6c.png">

## Visually:

Worth noting that, when searching for a pattern in the inserter, the pattern preview fly-outs do actually show both the title and the description. In this scenario, displaying the same text twice looks unnecessary and feels like an unpolished UI:

<img width="838" alt="Screenshot 2023-03-29 at 11 51 39" src="https://user-images.githubusercontent.com/1682452/228535375-091fa897-3bf6-48f3-8517-3a1deceb46a1.png">

One more example of title and description with the same text:

<img width="928" alt="Screenshot 2023-03-29 at 13 21 49" src="https://user-images.githubusercontent.com/1682452/228535564-1eb2a818-55b3-4027-9c94-5f713b056e80.png">

The repeated text looks off also when the description starts with the title text and then adds some more info:

<img width="878" alt="Screenshot 2023-03-29 at 11 52 17" src="https://user-images.githubusercontent.com/1682452/228536022-c5c8cdce-e3f2-452f-bd23-0a0ecc3bd27a.png">

I don't have permissions to edit on the patterns directory but I'm guessing the description text comes from the textarea in the editor sidebar:

<img width="1714" alt="Screenshot 2023-03-29 at 10 50 01" src="https://user-images.githubusercontent.com/1682452/228536332-cd188024-e649-4dbc-9253-7962bd2790eb.png">

Authors should take particular care when crafting the description text and always avoid repetition. Worth considering to clarify this point in the help text below the textarea or link to a page with some guidelines.

An example of better title and description is [this pattern](https://wordpress.org/patterns/pattern/three-columns-with-title-description-and-image/):
```
Title:       Three columns with title, description, and image
Description: Three columns. The first column has a title aligned to the bottom of the container, the second column has an italic description, and the third column has a portrait image.
```

In this case, the description actually describes the visual aspect of the pattern and it's meaningful enough, with little repetition.

## Impact on SEO

To my understanding, the author-provided title and description are used also for the Opengraph `og:title` and `og:description` / meta description on the patterns directory. See for example this page: https://wordpress.org/patterns/pattern/three-columns-of-text/

```
<meta property="og:title" content="Three columns of text" />
<meta property="og:description" content="Three columns of text." />
<meta name="description" content="Three columns of text." />
```

That's less than ideal. The repetition seems, at best, useless and, at worst, suboptimal for SEO. Search engines will tend to use the provided description which just repeats the info provided in the title, e.g.:

<img width="1010" alt="Screenshot 2023-03-29 at 13 57 27" src="https://user-images.githubusercontent.com/1682452/228540180-606c5da0-b42b-4bac-b7fd-560c549844c1.png">

**To Reproduce**
1. See the examples above.

**Expected behavior**
The pattern title and description to provide information that is actually useful for users.
- Title: a concise, descriptive name of the pattern.
- Description: a more detailed description of the pattern visual aspect.

Ideally, the information meant for SEO purposes should be provided separately. I'd recommend to consider to use two separate fields: SEO title and SEO description.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**
